contributed/matching_fof_mlz_blending.ipynb

```diff
@@ -0,0 +1,805 @@
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "scrolled": true
+   },
+   "outputs": [],
+   "source": [
+    "# Let's now turn to the truth catalog, turn of md5 sum check to save time\n",
+    "truth_cat = GCRCatalogs.load_catalog('dc2_truth_run1.2_static', {'md5': None})"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# for a reason that we will soon see, let's inspect the quantities in truth catalog\n",
+    "\n",
+    "print(sorted(truth_cat.list_all_quantities()))\n",
+    "print('---')\n",
+    "print(sorted(truth_cat.list_all_native_quantities()))"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# so we see there is not mag_i, but only mag_true_i (i.e., magnitude before lensing), and it maps to `i`\n",
+    "truth_cat.get_quantity_modifier('mag_true_i')"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# to make our `mag_filters` work, let's define mag_i for the truth catalog\n",
+    "truth_cat.add_quantity_modifier('mag_i', 'i')"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "scrolled": true
+   },
+   "outputs": [],
+   "source": [
+    "# get ra and dec from truth catalog\n",
+    "# note that we add i < 24.5 to the native filter to speed up load time\n",
+    "#truth_native_filters = (coord_filters + ['i < 24.5'])\n",
+    "truth_data = truth_cat.get_quantities(['ra', 'dec', 'object_id', 'star', 'sprinkled','agn','redshift','mag_true_i',\n",
+    "                                      'g','mag_true_g','mag_true_r', 'mag_true_u', 'mag_true_y', 'mag_true_z'],filters=star_sprinkled_filter)#, filters=mag_filters, native_filters=truth_native_filters)\n",
+    "\n",
+    "# We will use the object_id, star, and sprinkled columns when cross-referencing truth information with the extragalactic catalog."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# check number of objects in truth and coadd catalog with the cuts (filters) \n",
+    "print (len(coadd_data['ra']))\n",
+    "print (len(truth_data['ra']))"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# now we can really do the matching!\n",
+    "# FoFCatalogMatching.match takes a dictionary of catalogs to match, a friends-of-friends linking length. \n",
+    "# Because our \"catalog\" is not an astropy table or pandas dataframe, \n",
+    "# `len(truth_coord)` won't give the actual length of the table\n",
+    "# so we need to specify `catalog_len_getter` so that the code knows how to get the length of the catalog.\n",
+    "\n",
+    "results = FoFCatalogMatching.match(\n",
```

Comment:
The matching took a long (~20 min) time for me. Do you think we can make the catalogs smaller for the purposes of this tutorial, just to see that things work?

Reply:
we can use smaller catalogs to reduce running time. I will edit the file name!